components/navbar/CartButton.tsx

```diff
@@ -0,0 +1,45 @@
+import { faShoppingCart } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { CartPopover, useCart } from '@reservoir0x/reservoir-kit-ui'
+import { Flex, Button, Text } from 'components/primitives'
+
+const CartButton = () => {
+  const { data: cartItems } = useCart((cart) => cart.items)
+
+  return (
+    <CartPopover
+      trigger={
+        <Button
+          css={{
+            justifyContent: 'center',
+            width: '44px',
+            position: 'relative',
+          }}
+          size="small"
+          color="gray3"
+        >
+          <FontAwesomeIcon icon={faShoppingCart} width="16" height="16" />
+          {cartItems.length > 0 && (
+            <Flex
+              align="center"
+              justify="center"
+              css={{
+                borderRadius: '99999px',
+                width: 20,
+                height: 20,
+                backgroundColor: '$primary9',
+                position: 'absolute',
+                top: -8,
+                right: -6,
+              }}
+            >
+              <Text style="subtitle3">{cartItems.length}</Text>
```

Comment:
I think this text color should be white. It's hard to see the dark text on the purple background in light mode